src/Compilers/CSharp/Portable/Lowering/Instrumentation/DebugInfoInjector_SequencePoints.cs

```diff
@@ -104,24 +101,17 @@ private static TextSpan CreateSpan(SyntaxTokenList startOpt, SyntaxNodeOrToken s
         }
 
         private static int GetEndPosition(SyntaxNodeOrToken nodeOrToken)
-        {
-            if (nodeOrToken.IsToken)
-            {
-                return nodeOrToken.Span.End;
-            }
-            else
-            {
-                return nodeOrToken.AsNode().GetLastToken().Span.End;
-            }
-        }
+            => nodeOrToken.AsNode(out var node) ? node.GetLastToken().Span.End : nodeOrToken.Span.End;
 
         internal static void GetBreakpointSpan(VariableDeclaratorSyntax declaratorSyntax, out SyntaxNode node, out TextSpan? part)
         {
+            Debug.Assert(declaratorSyntax.Parent != null);
+
             var declarationSyntax = (VariableDeclarationSyntax)declaratorSyntax.Parent;
 
             if (declarationSyntax.Variables.First() == declaratorSyntax)
             {
-                switch (declarationSyntax.Parent.Kind())
+                switch (declarationSyntax.Parent!.Kind())
```

Comment:
>declarationSyntax.Parent! [](http://example.com/codeflow?start=24&length=25)

We prefer assertions unless there was an explicit check in the method which the compiler couldn't understand/track. This doesn't seem to be the case here.